### PR TITLE
Summary 2 on ramachandran2017searching

### PR DESCRIPTION
**Pull request by liaojh1998**
To be completed by tonight.

### CODE REVIEW

**Comment by liaojh1998** [comment]
Draft completed.

File: _summaries/ramachandran2017searching_2.md [hunk 1]
```diff
@@ -1,19 +1,63 @@
...
+
+    * _Activation Function Search_
+
+        The authors restricted the search space to functions composed from functions that takes in 1 or 2 scalar values and output a scalar. This is done by composing from multiple repetitions of the "core unit":
+
+        <p align="center">
+            <img src="https://d3i71xaburhd42.cloudfront.net/c8c4ab59ac29973a00df4e5c8df3773a3c59995a/2-Figure1-1.png" width="30%" height="30%">
+        </p>
+
+        Note that the inputs of the unary functions are restricted to the preactivation $$x$$ and the binary function outputs. This way, the activation function can always take a scalar value as input and output a single scalar value, like ReLU.
```
Comment (kraja99) [inline _summaries/ramachandran2017searching_2.md:24]: Maybe give a few examples of the unary or binary functions in the search space.

File: _summaries/ramachandran2017searching_2.md [hunk 2]
```diff
@@ -1,19 +1,63 @@
...
+
+        <p align="center">
+            <img src="https://d3i71xaburhd42.cloudfront.net/c8c4ab59ac29973a00df4e5c8df3773a3c59995a/3-Figure2-1.png" width="30%" height="30%">
+        </p>
+
+        The RNN predicts a single component of the activation function at each timestep, and each prediction is fed back as input to "predict" the component of the next timestep in an autoregressive fashion. This network is trained with reinforcement learning (Proximal Policy Optimization, [Schulman et al., 2017](https://arxiv.org/abs/1707.06347)) to output functions that have high validation accuracies by using validation accuracy as the reward. To reduce search time, the search algorithm batched candidate functions and aggregated their validation accuracies to update the search algorithm.
+
+    * _Search Evaluations_
+        
+        The searches are conducted by training ResNet-20 ([He et al., 2016a](https://ieeexplore.ieee.org/document/7780459)) on the CIFAR-10 ([Krizhevsky & Hinton, 2009](https://www.cs.toronto.edu/~kriz/learning-features-2009-TR.pdf)) by substituting each activation function instead of ReLU. Every network is trained for 10K steps.
```
Comment (kraja99) [inline _summaries/ramachandran2017searching_2.md:36]: Consider mentioning the use of "child networks" more explicitly, as well as the use of distributed training to parallelize computation.

**Review by kraja99 — APPROVED** [review]
Overall good summary on the paper!

File: _summaries/ramachandran2017searching_2.md (hunk 1 above)
Comment (liaojh1998) [reply _summaries/ramachandran2017searching_2.md]: Added all examples for reference.

File: _summaries/ramachandran2017searching_2.md (hunk 2 above)
Comment (liaojh1998) [reply _summaries/ramachandran2017searching_2.md]: Done!

**Comment by liaojh1998** [comment]
Changes:
* Added all examples of unary and binary functions.
* Made explicit the use of child networks, the fact that they took time to train, so that the authors added a distributed training scheme for search.
* Added comments on differences between ReLU and Swish, which models can take advantage of.

Let me know if there are any other issues. Thank you so much @kraja99 for the suggestions!

**Pull request merged**